src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -51,10 +53,18 @@ public boolean start() throws Exception {
 
     @Override
     public void onResume() {
-        super.onResume();
         setupTimer(System.currentTimeMillis());
     }
 
+    private TaskListener listener() {
+        try {
+            return getContext().get(TaskListener.class);
+        } catch (Exception x) {
+            LOGGER.log(Level.WARNING, null, x);
+            return TaskListener.NULL;
```

Comment:
Filed [JENKINS-40161](https://issues.jenkins-ci.org/browse/JENKINS-40161) to track the latter proposal.